### PR TITLE
Add SocketEngineType.DefaultTransport

### PR DESCRIPTION
**Pull request by antonfirsov**
So we can use the original `SocketsTransport` as a baseline

### CODE REVIEW

File: test/web/ConsoleLineArgumentsParser.cs [hunk 1]
```diff
@@ -11,15 +11,16 @@ public enum SocketEngineType
         EPoll,
         IOUring,
         IOUringTransport,
-        LinuxTransport
+        LinuxTransport,
+        DefaultSockets
```
Comment (tmds) [inline test/web/ConsoleLineArgumentsParser.cs]: Maybe drop `Default` from the name?

File: test/web/Program.cs [hunk 2]
```diff
@@ -34,38 +34,45 @@ public static IHostBuilder CreateHostBuilder(string[] args, CommandLineOptions c
...
-                    }
-                    else
-                    {
-                        webBuilder.UseLinuxAsyncSockets(options =>
+                            });
+                            break;
+                        case SocketEngineType.DefaultSockets:
+                            webBuilder.UseSockets(options =>
                             {
-                                options.DispatchContinuations = commandLineOptions.DispatchContinuations.Value;
```
Comment (tmds) [inline test/web/Program.cs:61]: @antonfirsov fyi, we've added an option for `DontAllocateMemoryForIdleConnections` in https://github.com/dotnet/aspnetcore/pull/19396.

**Review by tmds — APPROVED** [review]
LGTM, Thanks!

File: test/web/ConsoleLineArgumentsParser.cs (hunk 1 above)
Comment (antonfirsov) [reply test/web/ConsoleLineArgumentsParser.cs]: I'm unsure. In many cases such naming is used to indicate that a default option is being picked from the remaining enum members.

Comment (tmds) [reply test/web/ConsoleLineArgumentsParser.cs]: I know what you mean by default, though it is not the default here.
Maybe: KestrelDefaultTransport?
Anyway, feel free to pick what you like and merge this.

Comment (antonfirsov) [reply test/web/ConsoleLineArgumentsParser.cs]: Ended up with the name `DefaultTransport`, since other variants also lack the `Kestrel*` prefix.

**Pull request merged**